src/CoreApi/KeyApi.cs

```diff
@@ -1,91 +1,85 @@
-﻿using Ipfs.CoreApi;
-using Newtonsoft.Json.Linq;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Ipfs.CoreApi;
+using Newtonsoft.Json.Linq;
 
 namespace Ipfs.Http
 {
-    class KeyApi : IKeyApi
+    internal class KeyApi : IKeyApi
     {
         /// <summary>
         ///   Information about a local key.
         /// </summary>
-        public class KeyInfo : IKey
+        public sealed class KeyInfo : IKey
         {
+            public KeyInfo(MultiHash id, string name)
+            {
+                Id = id;
+                Name = name;
+            }
+
             /// <inheritdoc />
-            public MultiHash Id { get; set; }
+            public MultiHash Id { get; }
 
             /// <inheritdoc />
-            public string Name { get; set; }
+            public string Name { get; }
 
             /// <inheritdoc />
             public override string ToString()
             {
-                return Name;
+                return Name ?? string.Empty;

```

Comment:
Unnecessary fallback value.

```suggestion
                return Name;

```